chain/store/name.go

```diff
@@ -1,63 +1,174 @@
 package store
 
 import (
-	"strings"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/nknorg/nkn/common/serialization"
 )
 
-func getRegistrantId(registrant []byte) string {
-	return string(registrant)
+type nameInfo struct {
+	registrant []byte
+	expiresAt  uint32
 }
 
-func getNameId(name string) string {
-	return strings.ToLower(name)
+type namesCleanup map[string]struct{}
+
+func newNameInfo(registrant []byte, expiresAt uint32) *nameInfo {
+	return &nameInfo{registrant: registrant, expiresAt: expiresAt}
 }
 
-func (sdb *StateDB) updateName(registrant []byte, name string) error {
-	registrantId := getRegistrantId(registrant)
-	nameId := getNameId(name)
+func (ni *nameInfo) Serialize(w io.Writer) error {
+	if err := serialization.WriteVarBytes(w, ni.registrant); err != nil {
+		return err
+	}
+	if err := serialization.WriteUint32(w, ni.expiresAt); err != nil {
+		return err
+	}
+	return nil
+}
 
-	err := sdb.trie.TryUpdate(append(NamePrefix, nameId...), registrant)
+func (ni *nameInfo) Deserialize(r io.Reader) error {
+	var err error
+	ni.registrant, err = serialization.ReadVarBytes(r)
 	if err != nil {
 		return err
 	}
+	ni.expiresAt, err = serialization.ReadUint32(r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ni *nameInfo) Bytes() ([]byte, error) {
+	buff := bytes.NewBuffer(nil)
+	err := ni.Serialize(buff)
+	if err != nil {
+		return nil, err
+	}
+	return buff.Bytes(), nil
+}
 
-	return sdb.trie.TryUpdate(append(NameRegistrantPrefix, registrantId...), []byte(name))
+func (ni *nameInfo) String() string {
+	return fmt.Sprintf("%x %d", ni.registrant, ni.expiresAt)
```

Comment:
Should format `%064x` for got a fixed length `ni.registrant` in case 0 preamble.
e.g.  Should be `0000f352......`, not `f352......`.

Reply:
done